src/result-async.ts

```diff
@@ -0,0 +1,123 @@
+import { Result, Ok, Err } from './'
+import { isPromise } from './'
+
+export class ResultAsync<T, E> {
+  private _promise: Promise<Result<T, E>>
+
+  constructor(res: Promise<Result<T, E>>) {
+    this._promise = res
+  }
+
+  static fromPromise<T, E>(promise: Promise<T>): ResultAsync<T, E> {
+    return new ResultAsync(promise.then((value: T) => new Ok<T, E>(value)))
+  }
+
+  map<A>(_f: (t: T) => A | Promise<A>): ResultAsync<A, E> {
```

Comment:
change `_f` to `f` since it is being used in the body of this method. The convention that I've been abiding by is that if an argument is not being used then I signal it to the reader (and the compiler) by prepending the underscore.

Please follow this convention for all code in this PR.

Example:

https://github.com/supermacro/neverthrow/blob/master/src/result.ts#L77

Reply:
Of course, I misinterpreted your conventions. I fixed this.